packages/datepicker-react/src/DatePicker.tsx

```diff
@@ -150,25 +152,38 @@ export function DatePicker({
     }
 
     return (
-        <div className={componentClassName} ref={datepickerRef}>
-            <div className="jkl-datepicker__outer-wrapper">
-                <label className={inputClassName}>
-                    <Label variant={variant} forceCompact={forceCompact}>
-                        {label}
-                    </Label>
-                    <input
-                        placeholder={placeholder}
-                        type="text"
-                        aria-invalid={!!errorLabel}
-                        className={`jkl-text-field__input`}
-                        data-testid="jkl-datepicker__input"
-                        value={dateString}
-                        onChange={onInputChange}
-                        onFocus={openDatepicker}
-                    />
-                </label>
-
-                <div hidden={datepickerHidden}>
+        <div className={componentClassName}>
+            <Label htmlFor={uuid} standAlone variant={variant} forceCompact={forceCompact}>
+                {label}
+            </Label>
+            <div className={inputClassName}>
+                <input
+                    id={uuid}
+                    placeholder={placeholder}
+                    type="text"
+                    aria-invalid={!!errorLabel}
+                    className="jkl-text-field__input"
+                    data-testid="jkl-datepicker__input"
+                    value={dateString}
+                    onChange={onInputChange}
+                />
+                <button
+                    aria-label={calendarButtonTitle}
```

Comment:
I think using a visually hidden text inside the button is better than using aria-label:
See:
https://www.accessibility-developer-guide.com/knowledge/aria/bad-practices/
https://accessible360.com/accessible360-blog/use-aria-label-screen-reader-text/